packages/core/src/browser/shell/shell-layout-restorer.ts

```diff
@@ -101,6 +78,15 @@ export class ShellLayoutRestorer implements CommandContribution {
         }
     }
 
+    async restoreLayout(app: FrontendApplication): Promise<void> {
+        const serializedLayoutData = await this.storageService.getData<string>(this.storageKey);
+        if (serializedLayoutData === undefined) {
+            throw new Error('Layout data is not available.');
```

Comment:
Could we use a special error code to filter out real errors from this or have another contract?

Reply:
I'll change `layoutRestorer.restoreLayout()` to return `Promise<boolean>`.